stepfunction/validators.py

```diff
@@ -0,0 +1,32 @@
+import colander
+from kinto.core.errors import http_error, httpexceptions, ERRORS
+from kinto.core.storage.exceptions import RecordNotFoundError
+
+from .storage import get_record
+
+
+def record_validator(schema):
+    def validator(request, **kwargs):
+        try:
+            record = get_record(request)
+        except RecordNotFoundError:
+            details = {
+                "id": request.matchdict['id'],
+                "resource_name": "record"
+            }
+            response = http_error(httpexceptions.HTTPNotFound(),
+                                  errno=ERRORS.INVALID_RESOURCE_ID,
+                                  details=details)
+            raise response
+
+        try:
+            deserialized = schema.deserialize(record)
+        except colander.Invalid as e:
+            translate = request.localizer.translate
+            error_dict = e.asdict(translate=translate)
+            for name, msg in error_dict.items():
+                request.errors.add('body', name, msg)
```

Comment:
Maybe we should use a name that explicit that we are talking about the record body, because it seems like we are talking about the request body.